Cloneable/CloneableGenerator.cs

```diff
@@ -41,77 +76,131 @@ private void GenerateCloneMethods(GeneratorExecutionContext context)
             if (context.SyntaxReceiver is not SyntaxReceiver receiver)
                 return;
 
-            var classSymbols = GetClassSymbols(context, receiver);
+            var options = (context.Compilation as CSharpCompilation).SyntaxTrees[0].Options as CSharpParseOptions;
+            
+            var compilation = context.Compilation.AddSyntaxTrees(CSharpSyntaxTree.ParseText(SourceText.From(cloneableAttributeText, Encoding.UTF8), options)).
+                AddSyntaxTrees(CSharpSyntaxTree.ParseText(SourceText.From(clonePropertyAttributeText, Encoding.UTF8), options)).
+                AddSyntaxTrees(CSharpSyntaxTree.ParseText(SourceText.From(ignoreClonePropertyAttributeText, Encoding.UTF8), options));
+
+            cloneableAttribute = compilation.GetTypeByMetadataName("Cloneable.CloneableAttribute")!;
+            cloneAttribute = compilation.GetTypeByMetadataName("Cloneable.CloneAttribute")!;
+            ignoreCloneAttribute = compilation.GetTypeByMetadataName("Cloneable.IgnoreCloneAttribute")!;
+
+            var classSymbols = GetClassSymbols(compilation, cloneableAttribute, receiver);
             foreach (var classSymbol in classSymbols)
             {
-                context.AddSource($"{classSymbol.Name}_cloneable.cs", SourceText.From(CreateCloneableCode(classSymbol), Encoding.UTF8));
+                var attribute = classSymbol.GetAttributes().First(x => x.AttributeClass.Equals(cloneableAttribute));
```

Comment:
Use SymbolEqualityComparer.Default.Equals instead of Equals in order to fix the warning

Reply:
Seems like I missed this warning in Rider IDE... But I see it in VisualStudio.